src/event_form/steps/event-form-step3.controller.js

```diff
@@ -182,29 +187,48 @@ function EventFormStep3Controller(
 
   /**
    * Select location.
+   * @param {string} $id
+   * @param {string} $label
    * @returns {undefined}
    */
-  controller.selectLocation = function ($item, $model, $label) {
-
-    var selectedLocation = _.find($scope.locationsForCity, function (location) {
-      return location.id === $model;
-    });
-
-    // Assign selection, hide the location field and show the selection.
-    $scope.selectedLocation = selectedLocation;
-    $scope.locationAutocompleteTextField = '';
+  controller.selectLocation = function ($id, $label) {
 
-    var location = EventFormData.getLocation();
-    location.id = $model;
-    location.name = $label;
-    location.address = selectedLocation.address;
-    EventFormData.setLocation(location);
+    var selectedLocation = null;
+    if ($scope.isBookableEvent) {
+      // fetch the location based on the id when bookable event
+      return cityAutocomplete
+        .getPlacesById($id)
+        .then(function(locations) {
+          selectedLocation = locations[0];
```

Comment:
If you'd use `/place/{id}` and name the function `getPlaceById` (singular) you wouldn't have to do this `locations[0]` here.